provider/pkg/provider/provider.go

```diff
@@ -696,7 +697,11 @@ func (p *googleCloudProvider) Delete(_ context.Context, req *rpc.DeleteRequest)
 
 	uri := res.ResourceUrl(req.GetId())
 
-	if strings.HasSuffix(uri, ":getIamPolicy") {
+	// Setting the project IAM policy to empty is quiet destructive and it is hard to imagine when doing so
+	// on delete would be reasonable. Terraform provider resets the IAM policy as well but has strong warnings
+	// against deleting this resource.
+	if strings.HasSuffix(uri, ":getIamPolicy") && !strings.HasSuffix(resourceKey, ":ProjectIamPolicy") {
```

Comment:
Can we set `res.NoDelete` in metadata and respect it here?

Reply:
> Can we set `res.NoDelete` in metadata and respect it here?

We do set the noDelete (see next if block in the same file) but seem to special case the IAM policy resources to perform the reset anyway. I considered if we should just "abandon" no delete resources as is rather than reset them regardless. I think it would also be easier to explain to users and we can document it appropriately on the resource. The more I think about it, the more I prefer that mode. Thoughts?

Reply:
I have updated the PR to actually drop the special handling of IAM policy deletion in general and added documentation to reflect the same. Let me know if you have strong objections.